src/Dofs/sparsity_pattern.jl

```diff
@@ -102,38 +415,164 @@ function _coupling_to_local_dof_coupling(dh::DofHandler, coupling::AbstractMatri
     return outs
 end
 
-"""
-    _add_cross_coupling(coupling_sdh, dof_i, dof_j, cell_field_dofs, neighbor_field_dofs, i, j, sym, keep_constrained, ch, cnt, I, J)
+function _create_sparsity_pattern!(
+        sp::AbstractSparsityPattern, dh::DofHandler, ch::Union{ConstraintHandler, Nothing},
+        keep_constrained::Bool,
+        coupling::Union{Vector{<:AbstractMatrix{Bool}}, Nothing},
+        cross_coupling::Union{AbstractMatrix{Bool}, Nothing},
+        topology,
+    )
+    # 1. Add all connections between dofs for every cell while filtering based
```

Comment:
Maybe factor this out similar to `cross_element_coupling!` so it is possible to call it from user code building custom pattern?